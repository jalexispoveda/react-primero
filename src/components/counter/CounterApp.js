import React, { useState } from "react";
import PropTypes from "prop-types";

const CounterApp = (props) => {
  const { value } = props;
  const [counter, setCounter] = useState(value);

  const add = () => {
    setCounter(counter + 1);
  };

  const substract = () => {
    setCounter(counter - 1);
  };

  return (
    <React.Fragment>
      <h1>CounterApp</h1>
      <h2>{counter}</h2>
      <button onClick={substract}>-1</button>
      <button onClick={add}>+1</button>
      <button onClick={() => setCounter(0)}>RESET</button>
    </React.Fragment>
  );
};

CounterApp.propTypes = {
  value: PropTypes.number.isRequired,
};

export default CounterApp;
