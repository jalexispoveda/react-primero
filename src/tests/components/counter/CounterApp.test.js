import React from "react";
import CounterApp from "../../../components/counter/CounterApp";
import { shallow } from "enzyme/build";

describe("tests para counterApp", () => {
  const value = 100;
  let component = shallow(<CounterApp value={value}></CounterApp>);

  //Se ejecutara antes de cada prueba
  beforeEach(() => {
    component = shallow(<CounterApp value={value}></CounterApp>);
  });

  test("should show counterApp component ", () => {
    const value = 100;

    expect(component).toMatchSnapshot();
  });

  test("El contador deberia estar en 100", () => {
    const text = component.find("h2").text();

    expect(+text).toBe(value);
  });

  test("Debe incrementar contador cuando se haga click en +1 ", () => {
    component.find("button").at(1).simulate("click");

    const text = component.find("h2").text();

    expect(+text).toBe(101);
  });

  test("Debe decrementar contador cuando se haga click en -1 ", () => {
    component.find("button").at(0).simulate("click");

    const text = component.find("h2").text();

    expect(+text).toBe(99);
  });

  test("El buton reset deberia dejar el value en 0 ", () => {
    component.find("button").at(2).simulate("click");

    const text = component.find("h2").text();

    expect(+text).toBe(0);
  });
});
