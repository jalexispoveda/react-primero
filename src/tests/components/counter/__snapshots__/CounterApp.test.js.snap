// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests para counterApp should show counterApp component  1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2>
    100
  </h2>
  <button
    onClick={[Function]}
  >
    -1
  </button>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    RESET
  </button>
</Fragment>
`;
